### PR TITLE
Issue#4377 cpp-build-env Dockerfile 

### PR DESCRIPTION
**Pull request by tcsiwula**
re: Prepare docker image for building cpp-ethereum #4377 cpp-build-env Dockerfile. 
Not sure if this docker hub was pushed to or not: https://hub.docker.com/r/tcsiwula/cpp-build-env/ cc: @chfast https://github.com/ethereum/cpp-ethereum/issues/4377

### CODE REVIEW

File: scripts/docker/cpp-build-env.dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# build:          docker build -t test_build_20 .
+# run:            docker run -it --rm test_build_20:latest
+# versioning:     docker tag cpp-build-env ethereum/cpp-build-env:v0.8.29.2017
+# upload:         docker push tcsiwula/cpp-build-env
+
+# get latest stable debian release image -- see https://hub.docker.com/_/debian/
+from debian:latest
```
Comment (chfast) [inline scripts/docker/cpp-build-env.dockerfile]: 1. It should be `FROM`.
2. Do not use rolling version. Try sid or stretch.

File: scripts/docker/cpp-build-env.dockerfile [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+# maintainer email
+MAINTAINER Tim Siwula <tcsiwula@gmail.com>
```
Comment (chfast) [inline scripts/docker/cpp-build-env.dockerfile]: `MAINTAINER` is deprecated. Use `LABEL` (read the docs). And point it to C++ Ethereum developers team (no email at the moment, but you can link cpp-ethereum repo.

File: scripts/docker/cpp-build-env.dockerfile [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+# by defualt, docker runs all processes as root within the container
+USER root
```
Comment (chfast) [inline scripts/docker/cpp-build-env.dockerfile]: This is not needed unless we want to change the user. I think we should.

File: scripts/docker/cpp-build-env.dockerfile [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+RUN apt-get update -yq && \
+    apt-get install -yq software-properties-common && \
+    apt-get install -yq --no-install-recommends apt-utils && \
+    apt-get install -yq build-essential && \
+    apt-get install -yq libleveldb-dev && \
+    apt-get install -yq libmicrohttpd-dev && \
+# version cmake 3
+    apt-get install -yq cmake && \
+# version clang 3 -- unable to locate
+    apt-get install -yq clang && \
```
Comment (chfast) [inline scripts/docker/cpp-build-env.dockerfile]: Do not install clang at the moment. We need clang 4+ at least.

File: scripts/docker/cpp-build-env.dockerfile [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+# install gcc-6
+    apt-get install -yq gcc-6
```
Comment (chfast) [inline scripts/docker/cpp-build-env.dockerfile]: Let's install the default GCC only. And we need g++.

File: scripts/docker/cpp-build-env.dockerfile [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+# install gcc-7 -- unable to locate
+    #apt-get install -yq gcc-7-base
+
+ENTRYPOINT ["/bin/bash"]
```
Comment (chfast) [inline scripts/docker/cpp-build-env.dockerfile]: This is not needed?

File: README.md [hunk 7]
```diff
@@ -36,8 +36,7 @@ Windows          | [![AppVeyor](https://img.shields.io/appveyor/ci/ethereum/cpp-
 Git and GitHub is used to maintain the source code. Clone the repository by:
 
 ```shell
-git clone --recursive https://github.com/ethereum/cpp-ethereum.git
-cd cpp-ethereum
+git clone --recursive https://github.com/ethereum/cpp-ethereum.git && cd cpp-ethereum
```
Comment (chfast) [inline README.md]: None of the changes in this file make sense to end users. Please revert them.

File: scripts/docker/cpp-build-env.dockerfile (hunk 1 above)
Comment (tcsiwula) [reply scripts/docker/cpp-build-env.dockerfile]: Went for stretch :)

File: scripts/docker/cpp-build-env.dockerfile (hunk 2 above)
Comment (tcsiwula) [reply scripts/docker/cpp-build-env.dockerfile]: ^ good catch

File: scripts/docker/cpp-build-env.dockerfile (hunk 3 above)
Comment (tcsiwula) [reply scripts/docker/cpp-build-env.dockerfile]: For security reasons I assume, I changed it. I think most machines or the ones that I have used require  sudo for apt-install commands...

File: scripts/docker/cpp-build-env.dockerfile (hunk 4 above)
Comment (tcsiwula) [reply scripts/docker/cpp-build-env.dockerfile]: sounds good, figure out a clean way to do this soon.

File: scripts/docker/cpp-build-env.dockerfile (hunk 3 above)
Comment (tcsiwula) [reply scripts/docker/cpp-build-env.dockerfile]: What do you think about this, for best for this? @chfast https://askubuntu.com/questions/477551/how-can-i-use-docker-without-sudo

**Comment by tcsiwula** [comment]
Submitting a new PR.

**Pull request closed without merging**